src/Transform/BundleMemoryPools.cpp

```diff
@@ -518,13 +518,15 @@ class KrnlMoveConstantsUp : public OpRewritePattern<arith::ConstantOp> {
  *  Function pass that enables memory pooling for MemRefs.
  */
 
-class KrnlBundleMemoryPoolsPass
-    : public PassWrapper<KrnlBundleMemoryPoolsPass, OperationPass<FuncOp>> {
+class KrnlBundleMemoryPoolsPass : public PassWrapper<KrnlBundleMemoryPoolsPass,
+                                      OperationPass<func::FuncOp>> {
 
   BlockToMemPool blockToStaticPool;
   BlockToMemPool blockToDynamicPool;
 
 public:
+  MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(KrnlBundleMemoryPoolsPass)
```

Comment:
Could you just leave a comment to state what that macro is doing? I have seen it elsewhere added, might be useful to state how to add it elsewhere in the future when it will be needed

Reply:
When a pass is defined in an anonymous namespace, it requires an explicit TypeID.
For example, `MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID` should be placed within a public section of the declaration of the class.
Details in https://github.com/llvm/llvm-project/commit/5e50dd048e3a20cde5da5d7a754dfee775ef35d6.